modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts

```diff
@@ -145,6 +147,102 @@ export default class Ecdsa {
     return keyShare;
   }
 
+  /**
+   * Generate shares for participant at index and split keys `(threshold,numShares)` ways.
+   * @param {number} index participant index
+   * @param {number} threshold Signing threshold
+   * @param {number} numShares  Number of shares
+   * @param {Buffer} seed optional 64 byte seed to use for key generation
+   * @returns {UYShare} Returns share of private u value and commitment to public y value.
+   */
+  keyShareStep1(index: number, threshold: number, numShares: number, seed?: Buffer): UYShare {
+    if (!(index > 0 && index <= numShares && threshold <= numShares && threshold === 2)) {
+      throw 'Invalid KeyShare Config';
+    }
+
+    if (seed && seed.length < 64) {
+      throw new Error('Seed must have a length of at least 64 bytes');
+    }
```

Comment:
Nit: should have tests for these non-happy paths (throwing the errors)

Reply:
Added